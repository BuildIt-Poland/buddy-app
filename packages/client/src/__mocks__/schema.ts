export enum UserRole {
  Newbie = 'NEWBIE',
  Buddy = 'BUDDY',
  Talent = 'TALENT',
}

export enum TaskStatus {
  Completed = 'COMPLETED',
  Uncompleted = 'UNCOMPLETED',
}

export default {
  UserRole,
  TaskStatus,
};
