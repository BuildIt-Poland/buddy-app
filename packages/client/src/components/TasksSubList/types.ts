import { Task } from '@buddy-app/schema';
import { CheckboxProps } from '../TaskCheckbox';
import { TaskOptions } from '../TaskOptions';

export interface TasksSubListProps extends CheckboxProps, TaskOptions {
  tasks: Task[];
  tabIndex?: number;
  title?: string;
}
