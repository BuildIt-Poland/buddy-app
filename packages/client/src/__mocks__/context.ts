import { AuthState } from 'contexts/AuthContext';
import { UserRole } from '.';

export const mockedTalentContext = (context?: Partial<AuthState>) => ({
  isAuthenticated: true,
  loading: false,
  data: {
    role: UserRole.Talent,
    token: 'token',
    userId: '1234',
  },
  ...context,
});

export const mockedBuddyContext = (context?: Partial<AuthState>) => ({
  isAuthenticated: true,
  loading: false,
  data: {
    role: UserRole.Buddy,
    token: 'token',
    userId: '1234',
  },
  ...context,
});

export const mockedNewbieContext = (context?: Partial<AuthState>) => ({
  isAuthenticated: true,
  loading: false,
  data: {
    role: UserRole.Newbie,
    token: 'token',
    userId: '1234',
  },
  ...context,
});
