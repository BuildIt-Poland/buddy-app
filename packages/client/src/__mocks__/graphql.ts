import { BUDDY_MENU_DETAILS, NEWBIE_MENU_DETAILS } from 'graphql/user-menu.graphql';
import { AVATAR_HEADER } from 'graphql/avatar-header.graphql';
import { NEWBIE_SELECT } from 'graphql/newbie-select.graphql';
import { BUDDY_SELECT } from 'graphql/buddy-select.graphql';
import { TASK_DETAILS } from 'graphql/task-details.graphql';
import { ADD_NEWBIE_TASK } from 'graphql/add-task.graphql';
import { ADD_NEWBIE } from 'graphql/add-user.graphql';
import { GraphQLError } from 'graphql';

import {
  MutationAddNewbieTaskArgs,
  AuthPayload,
  TaskStatus,
  Newbie,
  QueryNewbieArgs,
  QueryBuddyArgs,
  UserRole,
} from '@buddy-app/schema';
import {
  NEWBIE_CONTACT_DETAILS,
  BUDDY_CONTACT_DETAILS,
  TALENT_CONTACT_DETAILS,
} from 'graphql/contact-details.graphql';
import { TASK_LIST } from 'graphql/task-list.graphql';
import { getBasicUserDetailsMock } from './general';

export const loginSuccessMock = (data?: Partial<AuthPayload>) => ({
  data: {
    login: {
      role: 'BUDDY',
      token: 'dummy-token',
      userId: '1',
    },
  },
});

export const loginFailedMock = () => new GraphQLError('No such user found');

export const loginNoNetworkMock = () => ({
  networkError: true,
});

export const NewbieAvatarDetails = (variables?: QueryNewbieArgs) => ({
  request: {
    query: AVATAR_HEADER,
    variables,
  },
  result: {
    data: {
      newbie: {
        id: '1',
        name: 'Tom Hanks',
        position: 'front-end',
        photo: 'some-url',
      },
    },
  },
});

export const tasksResponse = (filter?: TaskStatus) => {
  const tasks = [
    {
      id: '1',
      title: 'Call you head manager',
      description: '',
      createdAt: '',
      implementationDate: '',
      status: TaskStatus.Uncompleted,
      newbie: {} as Newbie,
    },
    {
      id: '2',
      title: 'Write you main manager',
      description: '',
      createdAt: '',
      implementationDate: '',
      status: TaskStatus.Completed,
      newbie: {} as Newbie,
    },
  ];

  switch (filter) {
    case TaskStatus.Completed:
      return [tasks[0]];
    case TaskStatus.Uncompleted:
      return [tasks[1]];
    default:
      return tasks;
  }
};

export const buddyMenuDetails = (variables?: Partial<QueryBuddyArgs>) => [
  {
    request: {
      query: BUDDY_MENU_DETAILS,
      variables,
    },
    result: {
      data: {
        buddy: {
          id: '1234',
          name: 'Tom Hanks',
          email: 'tom@wipro.com',
          allowPushedNotifications: true,
          photo: 'buddy-photo-url',
          newbies: [
            {
              id: 'id',
              name: 'Sandra Bullock',
              photo: 'newbie-photo-url',
            },
          ],
        },
      },
    },
  },
];

export const newbieMenuDetails = (variables?: Partial<QueryNewbieArgs>) => [
  {
    request: {
      query: NEWBIE_MENU_DETAILS,
      variables,
    },
    result: {
      data: {
        newbie: {
          id: '1234',
          name: 'Sandra Bullock',
          email: 'doejoe@wipro.com',
          allowPushedNotifications: true,
          photo: 'newbie-photo-url',
          buddy: {
            id: 'id',
            name: 'Tom Hanks',
            photo: 'newbie-photo-url',
          },
        },
      },
    },
  },
];

export const taskListResponse = (
  variables: Partial<QueryNewbieArgs>,
  newbie: Partial<Newbie>
) => [
  {
    request: {
      query: TASK_LIST,
      variables,
    },
    result: {
      data: {
        newbie,
      },
    },
  },
];

export const newbieContactDetails = (variables?: Partial<QueryNewbieArgs>) => [
  {
    request: {
      query: NEWBIE_CONTACT_DETAILS,
      variables,
    },
    result: {
      data: {
        newbie: getBasicUserDetailsMock(UserRole.Newbie),
      },
    },
  },
];

export const buddyContactDetails = (variables?: Partial<QueryBuddyArgs>) => [
  {
    request: {
      query: BUDDY_CONTACT_DETAILS,
      variables,
    },
    result: {
      data: {
        buddy: getBasicUserDetailsMock(UserRole.Buddy),
      },
    },
  },
];

export const talentContactDetails = (variables?: Partial<QueryBuddyArgs>) => [
  {
    request: {
      query: TALENT_CONTACT_DETAILS,
      variables,
    },
    result: {
      data: {
        buddy: getBasicUserDetailsMock(UserRole.Talent),
      },
    },
  },
];

export const newbieSelectMock = [
  {
    request: {
      query: NEWBIE_SELECT,
      variables: {
        buddyId: 'ck17sl83c9gya0b17dcvttzm4',
      },
    },
    result: {
      data: {
        buddy: {
          id: 'ck17sl83c9gya0b17dcvttzm4',
          name: 'Dummy',
          role: 'BUDDY',
          photo: null,
          newbies: [
            {
              id: 'ck17svulh9k2k0b17j31ansfk',
              photo: null,
              name: 'Test 2',
              position: 'Front End',
              startDate: null,
              newbieTasks: [
                {
                  id: '1',
                  status: TaskStatus.Completed,
                },
              ],
              buddyTasks: [
                {
                  id: '1',
                  status: TaskStatus.Uncompleted,
                },
              ],
            },
            {
              id: 'ck17swp2m9kcv0b17we0ibrdn',
              photo: null,
              name: 'Test 1',
              position: 'Front End',
              startDate: null,
              newbieTasks: [
                {
                  id: '1',
                  status: TaskStatus.Completed,
                },
              ],
              buddyTasks: [
                {
                  id: '1',
                  status: TaskStatus.Uncompleted,
                },
              ],
            },
          ],
        },
      },
    },
  },
];

export const buddySelectMock = [
  {
    request: {
      query: BUDDY_SELECT,
      variables: {
        talentId: 'ck17sl83c9gya0b17dcvttzm4',
      },
    },
    result: {
      data: {
        talent: {
          id: 'ck17sl83c9gya0b17dcvttzm4',
          name: 'Dummy',
          role: 'BUDDY',
          photo: null,
          buddies: [],
        },
      },
    },
  },
];

export const taskDetailsMock = [
  {
    request: {
      query: TASK_DETAILS,
      variables: {
        taskId: '1',
      },
    },
    result: {
      data: {
        task: {
          id: '1',
          title: 'New task',
          description: 'New task',
          status: TaskStatus.Uncompleted,
          newbie: getBasicUserDetailsMock(),
        },
      },
    },
  },
];

export const addTaskSuccessMock = (
  variables?: Partial<MutationAddNewbieTaskArgs>,
  data?: Partial<AuthPayload>
) => ({
  request: {
    query: ADD_NEWBIE_TASK,
    variables: {
      newbieId: '1234',
      input: {
        title: 'Test task',
        description: '<h1>Hello world!</h1>',
      },
      ...variables,
    },
  },
  result: {
    data: {
      addNewbieTask: {
        id: 'ck3sw1h1mqkd90964u4675qo2',
        title: 'Test task',
        newbie: {
          id: '1',
          newbieTasks: [],
        },
        ...data,
      },
    },
  },
});

export const addTaskFailedMock = (
  variables?: Partial<MutationAddNewbieTaskArgs>
) => ({
  request: {
    query: ADD_NEWBIE_TASK,
    variables: {
      newbieId: '1234',
      input: {
        title: 'Test task',
        description: '<h1>Hello world!</h1>',
      },
      ...variables,
    },
  },
  result: {
    errors: [new GraphQLError('No such user')],
  },
});

export const addNewbieSuccessMock = () => ({
  request: {
    query: ADD_NEWBIE,
    variables: {
      buddyId: '1234',
      input: {
        email: 'test_newbie@email.com',
        name: 'test newbie',
      },
    },
  },
  result: {
    data: {
      addNewbie: {
        id: 'ck3sw1h1mqkd90964u4675qo2',
        email: 'test_newbie@email.com',
        name: 'test newbie',
        role: UserRole.Newbie,
      },
    },
  },
});

export const addNewbieFailedMock = () => ({
  request: {
    query: ADD_NEWBIE,
    variables: {
      buddyId: '1234',
      input: {
        email: 'test_newbie@email.com',
        name: 'test newbie',
      },
    },
  },
  result: {
    errors: [new GraphQLError('No such user')],
  },
});
