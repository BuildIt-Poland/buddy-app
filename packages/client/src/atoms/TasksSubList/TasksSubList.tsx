import React from 'react';
import List from '@material-ui/core/List';
import ListSubheader from '@material-ui/core/ListSubheader';
import ListItem from '@material-ui/core/ListItem';
import { makeStyles } from '@material-ui/core/styles';
import TaskListItem from 'components/TaskListItem';
import { TasksSubListProps } from './types';

const useStyles = makeStyles(() => ({
  fullWidth: {
    width: '100%',
  },
}));

const TasksSubList: React.FC<TasksSubListProps> = ({
  tasks,
  title,
  tabIndex,
  onChange,
}) => {
  const { fullWidth } = useStyles();

  const ListHeader = (
    <ListSubheader disableSticky disableGutters>
      <strong>
        {title} ({tasks.length})
      </strong>
    </ListSubheader>
  );

  return (
    <ListItem disableGutters dense>
      <List dense className={fullWidth} subheader={ListHeader}>
        {tasks.map(task => (
          <TaskListItem
            key={task.id}
            task={task}
            onChange={onChange}
            tabIndex={tabIndex}
          />
        ))}
      </List>
    </ListItem>
  );
};

export default TasksSubList;
