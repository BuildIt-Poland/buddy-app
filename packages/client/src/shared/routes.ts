enum ROUTES {
  BASE = '/',
  LOGIN = '/login',
  FORGOT_PASSWORD = '/forgot-password',
  RESET_PASSWORD = '/reset-password/:token',
  ROUTE_404 = '/404',

  TALENT = '/talent',
  TALENT_SELECT_BUDDY = '/talent/buddies',
  TALENT_SELECT_NEWBIE = '/talent/buddies/:buddyId/newbies',
  TALENT_ADD_TASK = '/talent/newbies/:newbieId/add-task',
  TALENT_TASK_DETAILS = '/talent/newbies/:newbieId/tasks/:taskId',
  TALENT_TASKS_LIST = '/talent/newbies/:newbieId/tasks',
  TALENT_ADD_NEWBIE = '/talent/buddies/:buddyId/add-newbie',
  TALENT_ADD_BUDDY = '/talent/add-buddy',
  TALENT_ADD_TALENT = '/talent/add-talent',
  TALENT_DETAILS = '/talent/details',
  TALENT_BUDDY_DETAILS = '/talent/buddies/:buddyId/details',
  TALENT_NEWBIE_DETAILS = '/talent/newbies/:newbieId/details',

  BUDDY = '/buddy/',
  BUDDY_DETAILS = '/buddy/details',
  BUDDY_ADD_TASK = '/buddy/newbies/:newbieId/add-task',
  BUDDY_TASK_DETAILS = '/buddy/newbies/:newbieId/tasks/:taskId',
  BUDDY_TASKS_LIST = '/buddy/newbies/:newbieId/tasks',
  BUDDY_SELECT_NEWBIE = '/buddy/newbies',
  BUDDY_ADD_NEWBIE = '/buddy/add-newbie',
  BUDDY_NEWBIE_DETAILS = '/buddy/newbies/:newbieId/details',

  NEWBIE = '/newbie',
  NEWBIE_DETAILS = '/newbie/details',
  NEWBIE_TASKS_LIST = '/newbie/tasks',
  NEWBIE_TASK_DETAILS = '/newbie/tasks/:taskId',
  NEWBIE_BUDDY_DETAILS = '/newbie/buddy/:buddyId/details',
}

export { ROUTES };
