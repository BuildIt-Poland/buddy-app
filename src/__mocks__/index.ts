export * from './general';
export * from './graphql';
export * from './schema';
