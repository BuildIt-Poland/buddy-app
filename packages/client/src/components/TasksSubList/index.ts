import TasksSubList from './TasksSubList';
export * from './types';

export default TasksSubList;
