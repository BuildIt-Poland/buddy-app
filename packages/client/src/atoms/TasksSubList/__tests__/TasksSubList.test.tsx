import React from 'react';
import { create } from 'react-test-renderer';
import { MockedProvider } from '@apollo/react-testing';
import { MemoryRouter } from 'react-router';
import { ROUTES } from 'shared/routes';
import { taskDetailsMock } from '__mocks__';
import { tasksResponse, TaskStatus } from '__mocks__';
import TasksSubList from '..';

jest.mock('@material-ui/core/List', () => 'List');
jest.mock('@material-ui/core/ListSubheader', () => 'ListSubheader');
jest.mock('@material-ui/core/ListItem', () => 'ListItem');
jest.mock('components/TaskListItem', () => 'TaskListItem');

describe('Component - TasksSubList', () => {
  const render = (component: any) => (
    <MockedProvider mocks={taskDetailsMock} addTypename={false}>
      <MemoryRouter initialEntries={[ROUTES.BASE]}>{component}</MemoryRouter>
    </MockedProvider>
  );
  test(`renders correctly Todo section`, () => {
    const component = create(
      render(
        <TasksSubList title='Todo' tasks={tasksResponse(TaskStatus.Uncompleted)} />
      )
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  test(`renders correctly Done section`, () => {
    const component = create(
      render(
        <TasksSubList title='Done' tasks={tasksResponse(TaskStatus.Completed)} />
      )
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  test(`renders correctly without tasks and count`, () => {
    const component = create(render(<TasksSubList title='Todo' tasks={[]} />));
    expect(component.toJSON()).toMatchSnapshot();
  });
});
