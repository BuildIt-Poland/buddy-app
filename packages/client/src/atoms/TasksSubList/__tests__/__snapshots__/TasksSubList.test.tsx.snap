// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - TasksSubList renders correctly Done section 1`] = `
<ListItem
  dense={true}
  disableGutters={true}
>
  <List
    className="makeStyles-fullWidth-1"
    dense={true}
    subheader={
      <ListSubheader
        disableSticky={true}
      >
        <strong>
          Done
           (
          1
          )
        </strong>
      </ListSubheader>
    }
  >
    <TaskListItem
      task={
        Object {
          "createdAt": "",
          "description": "",
          "id": "1",
          "implementationDate": "",
          "newbie": Object {},
          "status": "UNCOMPLETED",
          "title": "Call you head manager",
        }
      }
    />
  </List>
</ListItem>
`;

exports[`Component - TasksSubList renders correctly Todo section 1`] = `
<ListItem
  dense={true}
  disableGutters={true}
>
  <List
    className="makeStyles-fullWidth-1"
    dense={true}
    subheader={
      <ListSubheader
        disableSticky={true}
      >
        <strong>
          Todo
           (
          1
          )
        </strong>
      </ListSubheader>
    }
  >
    <TaskListItem
      task={
        Object {
          "createdAt": "",
          "description": "",
          "id": "2",
          "implementationDate": "",
          "newbie": Object {},
          "status": "COMPLETED",
          "title": "Write you main manager",
        }
      }
    />
  </List>
</ListItem>
`;

exports[`Component - TasksSubList renders correctly without tasks and count 1`] = `
<ListItem
  dense={true}
  disableGutters={true}
>
  <List
    className="makeStyles-fullWidth-1"
    dense={true}
    subheader={
      <ListSubheader
        disableSticky={true}
      >
        <strong>
          Todo
           (
          0
          )
        </strong>
      </ListSubheader>
    }
  />
</ListItem>
`;
